addons/nuxeo-platform-3d/nuxeo-platform-3d-core/src/main/java/org/nuxeo/ecm/platform/threed/service/ThreeDBatchUpdateWork.java

```diff
@@ -92,6 +91,29 @@ public boolean isIdempotent() {
         return false;
     }
 
+    @Override
+    public boolean isGroupJoin() {
+        // This is a GroupJoin work with a trigger that can be used on the last work execution
+        return true;
+    }
+
+    @Override
+    public String getPartitionKey() {
+        return computeIdPrefix(repositoryName, docId);
+    }
+
+    @Override
+    public void onGroupJoinCompletion() {
+        if ("DONE".equals(getStatus())) {
```

Comment:
The status is supposed to be for human information, so using it for code logic is a bit dangerous.
And I only see a `setStatus("Done")`, not `"DONE"`, which shows why this is brittle. You should at least use a constant in both places to show commonality.
But why do you even need to check the status, given that `onGroupJoinCompletion()` seems to only be called if the work ended successfully?

Reply:
You are right it should be "Done" and I will use a constant. The reason is that there are many cases where the work returns without being in this "Done" state, to reproduce the existing behavior the event should be fired only for the last Work of the group that is in this Done state.